pinot-controller/src/main/java/org/apache/pinot/controller/recommender/realtime/provisioning/MemoryEstimator.java

```diff
@@ -430,44 +422,48 @@ private long calculateMemoryForCompletedSegmentsPerPartition(long completedSegme
     return _numSegmentsQueriedPerHost;
   }
 
+  public void cleanup() {
```

Comment:
Could we make this class auto-closable, so that we don't have to worry about where it needs to be called?

Reply:
+1

In any case we need to make the cleanup private so that user of MemoryEstimator should not worry about the protocol of cleaning up.

Reply:
Auto-closable will still be a public api - instead of calling .close() directly, the object needs to be used in try with resources.
In general, I agree with sidd to hide this from the consumer of this class.